app/src/main/java/com/keylesspalace/tusky/components/timeline/TiemlineViewModel.kt

```diff
@@ -0,0 +1,910 @@
+                Log.e(TAG, "Failed to load more at $position, wrong placeholder position")
+                return@launch
+            }
+            val placeholder = statuses[position].asPlaceholderOrNull() ?: run {
+                Log.e(TAG, "Not a placeholder at $position")
+                return@launch
+            }
+
+            val newViewData: StatusViewData = StatusViewData.Placeholder(placeholder.id, true)
+            statuses[position] = newViewData
+            triggerViewUpdate()
+
+            try {
+                val statuses = loadStatuses(
+                    fromStatus.id,
+                    toStatus.id,
+                    toMinusOne,
+                    TimelineRequestMode.NETWORK
+                )
+                replacePlaceholderWithStatuses(
+                    statuses.toMutableList(),
+                    isFullFetch(statuses),
+                    position
+                )
+            } catch (t: Exception) {
+                if (isExpectedRequestException(t)) {
+                    Log.e(TAG, "Failed to load gap", t)
+                    if (statuses[position] is StatusViewData.Placeholder) {
+                        statuses[position] = StatusViewData.Placeholder(placeholder.id, false)
+                    }
+                } else {
+                    throw t
+                }
+            }
+        }
+    }
+
+    fun reblog(reblog: Boolean, position: Int): Job = viewModelScope.launch {
+        val status = statuses[position].asStatusOrNull() ?: return@launch
+        try {
+            timelineCases.reblog(status.id, reblog).await()
+        } catch (t: Exception) {
+            ifExpected(t) {
+                Log.d(TAG, "Failed to reblog status " + status.id, t)
+
+            }
+        }
+    }
+
+    fun favorite(favorite: Boolean, position: Int): Job = viewModelScope.launch {
+        val status = statuses[position].asStatusOrNull() ?: return@launch
+
+        try {
+            timelineCases.favourite(status.id, favorite).await()
+        } catch (t: Exception) {
+            ifExpected(t) {
+                Log.d(TAG, "Failed to favourite status " + status.id, t)
+            }
+        }
+    }
+
+    fun bookmark(bookmark: Boolean, position: Int): Job = viewModelScope.launch {
+        val status = statuses[position].asStatusOrNull() ?: return@launch
+        try {
+            timelineCases.bookmark(status.id, bookmark).await()
+        } catch (t: Exception) {
+            ifExpected(t) {
+                Log.d(TAG, "Failed to favourite status " + status.id, t)
+            }
+        }
+    }
+
+    fun voteInPoll(position: Int, choices: List<Int>): Job = viewModelScope.launch {
+        val status = statuses[position].asStatusOrNull() ?: return@launch
+
+        val poll = status.status.poll ?: run {
+            Log.w(TAG, "No poll on status ${status.id}")
+            return@launch
+        }
+
+        val votedPoll = status.actionable.poll!!.votedCopy(choices)
```

Comment:
shouldn't this be `val votedPoll = poll.votedCopy(choices)`?

Reply:
Yes, this makes sense!